extensions/amp-3q-player/0.1/amp-3q-player.js

```diff
@@ -86,17 +83,42 @@ class Amp3QPlayer extends AMP.BaseElement {
     Services.videoManagerForDoc(el).register(this);
   }
 
-  /** @override */
-  layoutCallback() {
-    const iframe = createFrameFor(
-      this,
-      'https://playout.3qsdn.com/' +
+  /** private */
+  generateIframeSrc() {
+    const explicitParamsAttributes = ['key', 'timestamp', 'controls'];
+
+    let iframeSrc = 'https://playout.3qsdn.com/';
+
+    if (this.element.getAttribute(`data-datasource`)) {
+      iframeSrc +=
+        'config_by_metadata/' +
+        this.element.getAttribute(`data-project`) +
+        '/' +
+        this.element.getAttribute(`data-datafield`) +
+        '/' +
         encodeURIComponent(dev().assertString(this.dataId)) +
         // Autoplay is handled by VideoManager
-        '?autoplay=false&amp=true'
-    );
+        '?autoplay=false&amp=true';
+    } else {
+      iframeSrc +=
+        this.dataId +
+        // Autoplay is handled by VideoManager
+        '?autoplay=false&amp=true';
+    }
 
-    this.iframe_ = iframe;
+    explicitParamsAttributes.forEach(explicitParam => {
+      const val = this.element.getAttribute(`data-${explicitParam}`);
+      if (val) {
+        iframeSrc = addParamToUrl(iframeSrc, explicitParam, val);
+      }
+    });
+
+    return iframeSrc;
+  }
+
+  /** @override */
+  layoutCallback() {
+    this.iframe_ = this.generateIframeSrc();
```

Comment:
Does this work? `iframeSrc` generates a string, whereas an `<iframe>` was previously stored from `createFrameFor`.

Reply:
You we're right, corrected!